### DISCUSSION TITLE
How can we call update api endpoint.

### DISCUSSION THREAD

**Discussion by rachitpant** [ElMassimo/js_from_routes#31] [Q&A] [answered]
How can we call the update api endpoint. 
This is the generated endpoint stub 
```
// DO NOT MODIFY: This file was automatically generated by JsFromRoutes.
import { definePathHelper } from '@js-from-routes/axios'

export default {
  list: definePathHelper('get', '/api/tennis_ita_cs_maps'),
  update: definePathHelper('patch', '/api/tennis_ita_cs_maps/:ita_id'),
}
```
Also the documentation states its possible to customize the generated client stub. . Can you please provide some examples of how that can be done. It would be nice if we pass an object , and the path helper automatically extracts the path param and puts the rest of the params in body.

**Comment by rachitpant** [comment]
i missed the documentation about update.. its here: [Submitting data](https://js-from-routes.netlify.app/client/#submitting-data) , Still it would be nice to know how to customize template , so that the helper automatically extracts the path params and sends the rest. ( possibly even wrapping the params automatically in required params .  eg. for the above path it would be nice if everything is wrapped inside `tennis_ita_cs_map` so that the following works.
```
  def update
    TennisItaCsMap.find(params[:ita_id]).update!(permit_params)
  end

  private

  def permit_params
    params.require(:tennis_ita_cs_map).permit(:ita_id, :org_id)
  end
```
This is just an idea to further improve this helpful library.

**Comment by ElMassimo** [comment, marked as answer]
See [_Customizing the Generated Code_](https://js-from-routes.netlify.app/guide/codegen.html#customizing-the-generated-code-%F0%9F%9B%A0) for more info on how to customize the template, or the import path for `definePathHelper`.

You have full control on what the method does, for example, you could automatically wrap parameters.

---

When using this style of helper, I usually don't wrap the parameters in controller actions, instead preferring to do:

```ruby
  def permit_params
    params.permit(:ita_id, :org_id)
  end
```

In practice, I haven't found disadvantages for using this approach, although I do use controllers with few actions or permit parameters in specific actions for more granularity.
